api/app/GraphQL/Mutations/DeleteUserSkill.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\ExperienceSkill;
+use App\Models\UserSkill;
+
+final class DeleteUserSkill
+{
+    /**
+     * Soft delete a UserSkill model
+     * @param  null  $_
+     * @param  array{}  $args
+     * @return UserSkill
+     */
+    public function __invoke($_, array $args)
+    {
+        $existingModel = UserSkill::find($args['id']); // will never be null as policy check precedes this
```

Comment:
> will never be null as policy check precedes this

That might be true right now, but what if someone calls this function elsewhere, forgetting to run the policy? Or changes the policy? Better to cover your back and handle edge cases here.

Reply:
Sure, I can do that 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/7579/commits/e579e2fe9c3a60ac7ff588005e7036b2a39157f3